### PR TITLE
PERF-4256: Add missing window expresssions

### PR DESCRIPTION
**Pull request by projjal**
Thanks for submitting a PR to the Genny repo. Please include the following fields (if relevant) prior to submitting your PR.

**Jira Ticket:** PERF-4256

**Whats Changed:**  
Add workloads for the remaining window expressions: $addToSet, $integral, $push, $first, $last, $shift, $documentNumber, $rank, $denseRank

**Patch testing results:**  
[If applicable, link a patch test showing code changes running successfully](https://performance-monitoring-and-analysis.server-tig.prod.corp.mongodb.com/task/sys_perf_linux_standalone.2022_11_sliding_windows_patch_0b03b1ae353e85365589c8d4ad0c2d9433289d12_64877e35a4cf476a01f940bc_23_06_12_20_25_42/performanceData?time_series_filters=%5B%7B%22active%22%3Atrue%2C%22name%22%3A%22test%22%2C%22operator%22%3A%22contains%22%2C%22type%22%3A%22string%22%2C%22value%22%3A%22SlidingWindows.%22%7D%2C%7B%22active%22%3Atrue%2C%22name%22%3A%22measurement%22%2C%22operator%22%3A%22contains%22%2C%22type%22%3A%22string%22%2C%22value%22%3A%22%22%7D%2C%7B%22active%22%3Atrue%2C%22name%22%3A%22argString%22%2C%22operator%22%3A%22contains%22%2C%22type%22%3A%22string%22%7D%2C%7B%22active%22%3Atrue%2C%22name%22%3A%22patch_result%22%2C%22operator%22%3A%22eq%22%2C%22type%22%3A%22number%22%7D%2C%7B%22active%22%3Atrue%2C%22name%22%3A%22mainline_commit%22%2C%22operator%22%3A%22contains%22%2C%22type%22%3A%22string%22%7D%2C%7B%22active%22%3Atrue%2C%22name%22%3A%22mainline_result%22%2C%22operator%22%3A%22eq%22%2C%22type%22%3A%22number%22%7D%2C%7B%22active%22%3Atrue%2C%22name%22%3A%22mainline_std%22%2C%22operator%22%3A%22eq%22%2C%22type%22%3A%22number%22%7D%2C%7B%22active%22%3Atrue%2C%22name%22%3A%22percent%22%2C%22operator%22%3A%22eq%22%2C%22type%22%3A%22number%22%7D%2C%7B%22active%22%3Atrue%2C%22name%22%3A%22execution%22%2C%22operator%22%3A%22eq%22%2C%22type%22%3A%22number%22%7D%2C%7B%22active%22%3Atrue%2C%22name%22%3A%22z%22%2C%22operator%22%3A%22eq%22%2C%22type%22%3A%22number%22%7D%5D)

**Workload Submission form:**  
If applicable, only required if there is a new workload being added. Form can be found [here](https://docs.google.com/forms/d/1r0kOHvFUa7rJxVmX_wp9prxYU5K155yKyZ1y3d5yhZg/edit)

**Related PRs:**   
If applicable, link related PRs

PLEASE READ AND REMOVE THE SECTION BELOW BEFORE SUBMITTING YOUR PR FOR REVIEW

1.  Once (1) you get approval from the STM or the Product Perf team for your change and (2) all your CI tests pass, you can initiate a merge by entering a comment on your PR with the text

    > evergreen merge

    This kicks your PR into the [evergreen commit queue][ecq]. Evergreen will automatically squash-merge your commit after checking that you have PR approval and that the CI tests all pass.

[ecq]: https://github.com/evergreen-ci/evergreen/wiki/Commit-Queue

### CODE REVIEW

File: src/workloads/query/SlidingWindows.yml [hunk 1]
```diff
@@ -127,6 +127,189 @@ Actors:
...
+            partitionBy: "$partitionKey",
+            sortBy: {time: 1},
+            output: {shifted: {
+              $shift: {
+                  output: "$temp",
+                  by: 500,
+                  default: "Not available"
+               }}}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: DocumentNumber
```
Comment (lriuui0x0) [inline src/workloads/query/SlidingWindows.yml:280]: $documentNumber/$rank/$denseRank probably makes more sense in the CumulativeWindows.yml.

**Review by lriuui0x0 — COMMENTED** [review]
Can we also add a workload to test complex partitionBy expression in a separate file? That test will particularly exercise the combination of stage builders once the pushdown of those pipelines is enabled.

File: src/workloads/query/SlidingWindows.yml [hunk 2]
```diff
@@ -127,12 +127,175 @@ Actors:
...
+            partitionBy: "$partitionKey",
+            sortBy: {time: 1},
+            output: {shifted: {
+              $shift: {
+                  output: "$temp",
+                  by: 500,
+                  default: "Not available"
+               }}}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: MovingAvgPositionBasedWithComplexPartition
```
Comment (lriuui0x0) [inline src/workloads/query/SlidingWindows.yml:280]: We can remove this now that there's a separate file.

**Review by lriuui0x0 — APPROVED** [review]
Approve with one suggestion of removal duplicate tests

**Comment by projjal** [comment]
[sys-perf build](https://spruce.mongodb.com/version/648b010a57e85a4b36baa507/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)

File: src/workloads/query/SlidingWindows.yml [hunk 3]
```diff
@@ -127,6 +129,19 @@ Actors:
               $derivative: {input: "$temp", unit: "second"},
               window: {documents: [-500, 0]}}}}}]
         cursor: {batchSize: *batchSize}
+  - *Nop
+
+
+- Name: SlidingWindows
```
Comment (lriuui0x0) [inline src/workloads/query/SlidingWindows.yml:135]: Why does separating to a new phase work?

Comment (projjal) [reply src/workloads/query/SlidingWindows.yml:135]: I did it to not contaminate the operation throughput of earlier existing tests. Apparently the operation throughput is the number of operations per unit of phase time and not per unit of that operation time.

**Comment by projjal** [comment]
evergreen merge

**Pull request merged**